pkg/endpoint/policy.go

```diff
@@ -80,139 +69,49 @@ func getSecurityIdentities(labelsMap *identityPkg.IdentityCache, selector *api.E
 	return identities
 }
 
-func (e *Endpoint) sweepFilters(oldPolicy *policy.L4Policy,
-	identities *identityPkg.IdentityCache) (errors int) {
-
-	oldEntries, err := e.PolicyMap.DumpToSlice()
-	if err != nil {
-		e.getLogger().WithError(err).WithFields(logrus.Fields{
-			logfields.PolicyRevision: oldPolicy.Revision,
-		}).Warning("Delete stale l4 policy failed")
-		errors++
-		return
-	}
-	for _, entry := range oldEntries {
-		id := identityPkg.NumericIdentity(entry.Key.GetIdentity())
-		if _, ok := (*identities)[id]; !ok {
-			if err := e.PolicyMap.DeleteEntry(&entry); err != nil {
-				e.getLogger().WithError(err).WithFields(logrus.Fields{
-					logfields.PolicyRevision: oldPolicy.Revision,
-					logfields.Identity:       id,
-				}).Warning("Delete stale l4 policy failed")
-				errors++
-			}
-		}
-	}
-
-	return
-}
-
-// removeOldFilter removes the old l4 filter from the endpoint.
-// Returns a map that represents all policies that were attempted to be removed;
-// it maps to whether they were removed successfully (true or false)
-func (e *Endpoint) removeOldFilter(identities *identityPkg.IdentityCache,
-	filter *policy.L4Filter, direction policymap.TrafficDirection) {
-
-	port := uint16(filter.Port)
-
-	for _, sel := range filter.Endpoints {
-		for _, id := range getSecurityIdentities(identities, &sel) {
-			srcID := id.Uint32()
-			if err := e.PolicyMap.Delete(srcID, port, filter.U8Proto, direction); err != nil {
-				// This happens when the policy would add
-				// multiple copies of the same L4 policy. Only
-				// one of them is actually added, but we'll
-				// still try to remove it multiple times.
-				e.getLogger().WithError(err).WithFields(logrus.Fields{
-					logfields.L4PolicyID:       srcID,
-					logfields.TrafficDirection: direction,
-				}).Debug("deletion of old L4 policy failed")
-
-			}
-		}
-	}
-}
-
-// applyNewFilter adds the given l4 filter to the endpoint.
-// Returns the number of errors that occurred while when applying the
-// policy.
-// Applies for L3-dependent L4, and not for L4-only policy.
-func (e *Endpoint) applyNewFilter(identities *identityPkg.IdentityCache,
-	filter *policy.L4Filter, direction policymap.TrafficDirection) int {
+func (e *Endpoint) convertL4FilterToPolicyMapKeys(identities *identityPkg.IdentityCache,
```

Comment:
`IdentityCache` is a map, so it already has reference semantics. Don't pass it as a pointer.

Reply:
Fixed.